frontend/src/app/modules/bim/bcf/bcf-viewer-bridge/revit-bridge.service.ts

```diff
@@ -49,21 +56,23 @@ export class RevitBridgeService extends ViewerBridgeService {
         filter(message => message.messageType === 'ViewpointData' && message.trackingId === trackingId),
         first()
       )
-      .toPromise()
-      .then((message) => {
-        let viewpointJson = JSON.parse(message.messagePayload);
-
-        viewpointJson.snapshot = {
-          snapshot_type: 'png',
-          snapshot_data: viewpointJson.snapshot
-        };
-
-        return viewpointJson;
-      });
+      .pipe(
+        map((message) => {
+          let viewpointJson = JSON.parse(message.messagePayload);
+
+          viewpointJson.snapshot = {
+            snapshot_type: 'png',
+            snapshot_data: viewpointJson.snapshot
+          };
+          return viewpointJson;
+        })
+      )
   }
 
-  showViewpoint(data:BcfViewpointInterface) {
-    this.sendMessageToRevit('ShowViewpoint', this.newTrackingId(), JSON.stringify(data));
+  public showViewpoint(workPackage:WorkPackageResource, index:number) {
+     this.viewpointsService
+              .getViewPoint$(workPackage, index)
```

Comment:
We should either write "viewpoint" or "viewPoint" but should not mix it. And we did write it "viewpoint" so far.